dbdimp.c

```diff
@@ -2386,6 +2386,18 @@ static bool mariadb_dr_connect(
       imp_dbh->sock_fd = sock_os;
 #endif
 
+#ifdef _WIN32
+      {
+        /*
+         * Winsock SO_UPDATE_CONNECT_CONTEXT option needs to be set on the socket,
+         * otherwise getpeername, getsockname, getsockopt, setsockopt, and shutdown
+         * functions would fail with WSAENOTCONN error.
+         */
+        DWORD value = 1;
+        setsockopt(imp_dbh->sock_fd, SOL_SOCKET, SO_UPDATE_CONNECT_CONTEXT, (const char *)&value, sizeof(value));
```

Comment:
Do I understand it correctly that this fixes the appveyor errors from the previous commit?

Reply:
yes